### PR TITLE
[PBNTR-184] Add Status Flag to Menu.yml

### PR DESCRIPTION
**Pull request by nidaqg**
**What does this PR do?**

[RUNWAY STORY](https://nitro.powerhrg.com/runway/backlog_items/PBNTR-184)

This PR will:

- ✅ Add a 'status' flag to menu.yml file to track kit status ('stable' or 'beta')
- ✅ Create new aggregate_kits method to include kit status
- ✅ Use new method to render sidebar so if 'beta' kit, does not show up in sidebar
- ✅ Edit search_list method to exclude 'beta' kits from typeahead search
- ✅ kit_show page to use new method (call out on top if kit beta)
- ✅ category page to use new method (do not show if kit beta)
- ✅ all kits page to use new method (do not show if kit beta)

**Screenshots:** 

**How to test?** Steps to confirm the desired behavior:
1. Test search typeahead
2. Test all kits page
3. Test category page
4. Test kit show pages
5. Check sidebar

### CODE REVIEW

File: playbook-website/app/javascript/components/MainSidebar/NavComponents/KitsNavComponent.tsx [hunk 1]
```diff
@@ -160,24 +160,18 @@ export const KitsNavItem = ({
...
+            cursor="pointer"
+            dark={dark}
+            fontSize="small"
+            key={`${sublink}-${j}`}
+            link={generateLink(categoryKey, sublink, type)}
+            marginY="none"
+            onClick={() => handleSubItemClick(j, sublink, kitIndex)}
+            paddingY="xxs"
+            text={linkFormat(sublink)}
+          />
```
Comment (nidaqg) [inline playbook-website/app/javascript/components/MainSidebar/NavComponents/KitsNavComponent.tsx:174]: this is just removing the 'hack' we put into place [on this commit](https://github.com/powerhome/playbook/pull/3084/commits/309edb686610fc190012be7beb1feb8a3c592608) on PBNTR-177

File: playbook-website/app/javascript/components/MainSidebar/index.tsx [hunk 2]
```diff
@@ -10,19 +10,34 @@ const MainSidebar = ({
...
+    });
+  };
+  
+  const kits = transformKitsData(kits_with_status);
+  
   //hook into collapsible logic for all components nested nav items
   const collapsibles = kits.map(() => useCollapsible());
 
-  const filteredSearchList = search_list.filter(obj => obj.label !== "Advanced Table");
-
```
Comment (nidaqg) [inline playbook-website/app/javascript/components/MainSidebar/index.tsx:25]: Again, removing the hack put into place on PBNTR-177

File: playbook-website/app/javascript/components/MainSidebar/index.tsx [hunk 3]
```diff
@@ -55,7 +70,7 @@ const MainSidebar = ({
         <KitSearch
           classname="desktop-kit-search"
           id="desktop-kit-search"
-          kits={filteredSearchList}
+          kits={search_list}
```
Comment (nidaqg) [inline playbook-website/app/javascript/components/MainSidebar/index.tsx:73]: Again, removing the hack put into place on PBNTR-177

File: playbook-website/lib/pb_doc_helper.rb [hunk 4]
```diff
@@ -57,12 +56,10 @@ def get_kits_pb_website
...
-      if kit != "advanced_table"
-        title = pb_doc_render_clickable_title(kit, type)
-        ui = raw("<div class='pb--docItem-ui'>
-            #{pb_kit(kit: kit, type: type, show_code: code, limit_examples: limit_examples, dark_mode: dark_mode)}</div>")
-        title + ui
-      end
+      title = pb_doc_render_clickable_title(kit, type)
+      ui = raw("<div class='pb--docItem-ui'>
+          #{pb_kit(kit: kit, type: type, show_code: code, limit_examples: limit_examples, dark_mode: dark_mode)}</div>")
+      title + ui
```
Comment (nidaqg) [inline playbook-website/lib/pb_doc_helper.rb:62]: Again, removing the hack put into place on PBNTR-177

**Review by thestephenmarshall — APPROVED** [review]

**Pull request merged**